lib/database/repo.ex

```diff
@@ -1,3 +1,16 @@
+defmodule MethodMissing do
+  defmacro __using__(_opts) do
```

Comment:
dialyzer does not like it:

```
Starting Dialyzer
[
  check_plt: false,
  init_plt: '/home/user/repo/_build/dev/dialyxir_erlang-24.0.1_elixir-1.12.0_deps-dev.plt',
  files: ['/home/user/repo/_build/dev/lib/bors/ebin/Elixir.BorsNG.GitHub.Server.beam',
   '/home/user/repo/_build/dev/lib/bors/ebin/Elixir.BorsNG.ErrorView.beam',
   '/home/user/repo/_build/dev/lib/bors/ebin/Elixir.BorsNG.Database.AttemptState.beam',
   '/home/user/repo/_build/dev/lib/bors/ebin/Elixir.BorsNG.GitHub.FriendlyMock.beam',
   '/home/user/repo/_build/dev/lib/bors/ebin/Elixir.BorsNG.Router.Helpers.beam',
   ...],
  warnings: [:no_unused, :error_handling, :race_conditions, :unknown]
]
Total errors: 142, Skipped: 0, Unnecessary Skips: 0
done in 0m4.93s
lib/database/context/dashboard.ex:0:call_to_missing
Call to missing or private function BorsNG.Database.Repo.all/1.
________________________________________________________________________________
lib/database/context/dashboard.ex:0:call_to_missing
Call to missing or private function BorsNG.Database.Repo.all/1.
__________________________________________________________________
```

the alternative is to use `BorsNG.Application.fetch_repo()` for all `Repo.` calls in the repository

what do you think?

Reply:
Could we use wrapper methods? Perhaps generated by a macro?

```
def all(q): BorsNG.Application.fetch_repo().all(q)
def all(q, opts): BorsNG.Application.fetch_repo().all(q, opts)
def get(q, id, opts): BorsNG.Application.fetch_repo().get(q, id, opts)
def get!(q, id, opts): BorsNG.Application.fetch_repo().get(q, id, opts)
... etc
```

Reply:
@notriddle we added a macro to define the functions dynamically from a file